### ISSUE TITLE
Certain search results append "#backdrop" to URL when page loads

### ISSUE THREAD

**Issue by hillaryfraley** [sensu/sensu-docs#2287]
### Description
When I open certain links from the search bar on our docs site, #backdrop is appended to the URL.

I don't know if there is something special about the pages that open this way -- it doesn't happen to every page.
**edit: see https://github.com/sensu/sensu-docs/issues/2287#issuecomment-887610921 below**

See the attached GIF.

### Category

This is an issue with:

- [x] bug (bug in site build, layout, or functionality)
- [ ] enhancement (add new or updated site functionality)

### Affected pages
Affects docs search

### Expected behavior
The only hashtagged text in the URL should be headings on the page

### Current behavior
1. Search Go docs for `mtls`
2. Click result "Secrets"
3. Page loads with `#backdrop` in URL

### Changes needed
Unknown

### Screenshots
![backdrop-tag](https://user-images.githubusercontent.com/19190208/77006367-572af080-6930-11ea-88f6-f001f3b6b914.gif)

### Additional context
Possibilities to test:
- Happens when the result is just a page (like https://docs.sensu.io/sensu-go/latest/reference/secrets/) rather than a specific section on a page (like https://docs.sensu.io/sensu-go/latest/reference/secrets/#secret-specification).
- ~It's a non-word abbreviation thing [Tested this: etcd and tls do not produce the same result]~
- It's only terms that do not appear in any headings inside a doc -- terms that are only in regular text
- It's only terms that appear only once on a page

**Comment by hillaryfraley** [comment]
I searched the [Algolia docs](https://www.algolia.com/doc/) for "backdrop" without results.

I also searched our docs repo for "backdrop" -- we do have a #backdrop style class: https://github.com/sensu/sensu-docs/search?q=backdrop&unscoped_q=backdrop

**Comment by hillaryfraley** [comment]
I think our updates in https://github.com/sensu/sensu-docs/pull/2234 must have caused some change that resolved this odd #backdrop addition, at least for the one document where we knew it was happening.

Edit: incorrect - searching for "mtls" still retrieves the Secrets reference and appends `#backdrop` to the url.

Maybe try removing parenthetical inside link bracket

**Comment by hillaryfraley** [comment]
Found another instance: search for "sensuctl dump" and click the result for https://docs.sensu.io/sensu-go/latest/sensuctl/back-up-recover/. When the page loads, the URL has `#backdrop` appended.

**Comment by hillaryfraley** [comment]
As of 6.0 docs, search for `mtls` doesn't include the secrets reference in the results. It's still there for 5.21 and 5.20, and the URL still appends `#backdrop`. We moved the secrets reference as of 6.0, but the 6.0 and 6.1 docs do still include `mTLS` on the page. So now I'm wondering how moving the doc could result in removing the `#backdrop` thing in the URL.

The sensuctl dump instance I found still has `#backdrop` appended to the URL for all versions.

**Comment by hillaryfraley** [comment]
I found another one (search term == `bsm`), and I think I know what causes a URL with the `#backdrop` appendix vs not.

I believe Algolia appends the `#backdrop` to the URL for search results that point to a top-level page rather than a section on a page. If a search term returns a result that is not within a section, the result title is identical to the page title. If you click that result, the URL will include `#backdrop`.

Algolia docs search repo has a PR that references postcss autoprefixer (see https://github.com/algolia/docsearch/pull/559), which has some code that uses `backdrop`:
https://github.com/postcss/autoprefixer/blob/main/lib/hacks/backdrop-filter.js
https://github.com/postcss/autoprefixer/blob/efca4fae67068ca2e5e840a8926a42e2deb9eff7/data/prefixes.js#L258

In short, since I understand what's happening now and I don't think we can change it to eliminate the `#backdrop` appendix from these search URLs, I think it's OK to close this issue.

**Comment by hillaryfraley** [comment]
Found another one: search for `limited service account` and the results include one top-level result with matching doc and result titles: https://docs.sensu.io/sensu-go/latest/operations/control-access/create-limited-service-accounts/#backdrop
